portfolio/src/main/webapp/script.js

```diff
@@ -37,18 +37,23 @@ function login(url){
   window.location = url;
 }
 
-function logout(url){
-  window.location = url;
-}
-
-async function checkLoginStatus(){
+async function setButtonName(){
   const response = await fetch('/login');
   const status = await response.text();
-  
-  if(status.localeCompare("/_ah/login?continue=%2Findex.html")){
-      login(status);
+  let obj = JSON.parse(status);
+  let element = document.getElementById("login-button");
+  if(!(obj.status.localeCompare("Login"))){
```

Comment:
Since obj.status will either be Login or Logout, any reason you don't just want to do this:
element.value = obj.status;